lib/puppet/type/apache_directive.rb

```diff
@@ -26,6 +26,14 @@ def insync?(is)
     end
   end
 
+  newparam(:namevar) do
```

Comment:
I'd rather user `name` than `namevar` for that.

I know this might change the behavior, so I'd suggest something like this:

* use `:name` as the default namevar
* add a `:directive` parameter for the directive name
* feed both `:name` and `:directive` in `self.title_patterns` with respectively the full match and the directive only
* have `:directive` default to `self[:name]` for compatibility purposes

This should make it backwards-compatible, and provider a nicer interface imo.